src/OrchardCore/OrchardCore.Setup.Core/SetupService.cs

```diff
@@ -89,60 +89,57 @@ public async Task<IEnumerable<RecipeDescriptor>> GetSetupRecipesAsync()
         /// <inheritdoc />
         public async Task<string> SetupAsync(SetupContext context)
         {
+            string executionId = null;
             var initialState = context.ShellSettings.State;
             try
             {
-                var executionId = await SetupInternalAsync(context);
+                // Set the shell state to 'Initializing' to guarantee that any subsequent HTTP requests
+                // will result in a 'Service Unavailable' response while the tenant setup is in progress.
+                context.ShellSettings.AsInitializing();
```

Comment:
I suggest to keep the code of all this method as it was before.

Reply:
It's very weird/not-readable to change the status into 2 methods. What I made here will altering the state of the same object in the same method.

Reply:
This method reverts the state without having to know which state will be set in the other method. Yes maybe this separation is not needed in the same class, I don't know why it was done like this, maybe these 2 methods were initially in 2 separate services.

 I already thought about it but I just kept the code unchanged. Otherwise we could also say that we can only setup an `Uninitialized` tenant and then the `initialState` becomes useless. So I'm not against but in a separate PR with maybe more cleanups.